src/Installer.php

```diff
@@ -255,6 +316,38 @@ private function downloadFile(HttpClient $httpClient, $url, $targetPath)
     }
 
     /**
+     * Downloads the available puli versions.
+     *
+     * @param HttpClient $httpClient The client to use.
+     * @param string     $url        The URL to download.
+     *
+     * @return array|null The available versions, null if the download failed.
+     *
+     * @throws RuntimeException If an error occurs.
+     */
+    public function downloadVersions(HttpClient $httpClient, $url)
+    {
+        ErrorHandler::register();
+
+        $versions = $httpClient->download($url);
+
+        if (ErrorHandler::hasErrors()) {
```

Comment:
If we just move it after the `unregister()` statement the `json_decode()` errors won't be detected because they are handled by `json_last_error()` instead of the `ErrorHandler`.
Actually the `usort()` call will generate an error if the json is malformed.